client/src/components/ElicitationModal.tsx

```diff
@@ -0,0 +1,181 @@
+import Ajv from "ajv";
+
+export interface ElicitationRequest {
+  id: number;
+  message: string;
+  requestedSchema: JsonSchemaType;
+  resolve: (response: ElicitationResponse) => void;
+}
+
+export interface ElicitationResponse {
+  action: "accept" | "reject" | "cancel";
+  content?: Record<string, unknown>;
+}
+
+interface ElicitationModalProps {
+  request: ElicitationRequest | null;
+  onClose: () => void;
+}
+
+const ElicitationModal = ({ request, onClose }: ElicitationModalProps) => {
+  const [formData, setFormData] = useState<JsonValue>({});
+  const [validationError, setValidationError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (request) {
+      const defaultValue = generateDefaultValue(request.requestedSchema);
+      setFormData(defaultValue);
+      setValidationError(null);
+    }
+  }, [request]);
+
+  if (!request) return null;
+
+  const validateEmailFormat = (email: string): boolean => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const validateFormData = (
+    data: JsonValue,
+    schema: JsonSchemaType,
+  ): boolean => {
+    if (
+      schema.type === "object" &&
+      schema.properties &&
+      typeof data === "object" &&
+      data !== null
+    ) {
+      const dataObj = data as Record<string, unknown>;
+
+      if (Array.isArray(schema.required)) {
+        for (const field of schema.required) {
+          const value = dataObj[field];
+          if (value === undefined || value === null || value === "") {
+            setValidationError(`Required field missing: ${field}`);
+            return false;
+          }
+        }
+      }
+
+      for (const [fieldName, fieldValue] of Object.entries(dataObj)) {
+        const fieldSchema = schema.properties[fieldName];
+        if (
+          fieldSchema &&
+          fieldSchema.format === "email" &&
+          typeof fieldValue === "string"
+        ) {
+          if (!validateEmailFormat(fieldValue)) {
+            setValidationError(`Invalid email format: ${fieldName}`);
+            return false;
+          }
+        }
+      }
+    }
+
+    return true;
+  };
+
+  const handleAccept = () => {
+    try {
+      if (!validateFormData(formData, request.requestedSchema)) {
+        return;
+      }
+
+      const ajv = new Ajv();
+      const validate = ajv.compile(request.requestedSchema);
+      const isValid = validate(formData);
+
+      if (!isValid) {
+        const errorMessage = ajv.errorsText(validate.errors);
+        setValidationError(errorMessage);
+        return;
+      }
+
+      request.resolve({
+        action: "accept",
+        content: formData as Record<string, unknown>,
+      });
+      onClose();
+    } catch (error) {
+      setValidationError(
+        error instanceof Error ? error.message : "Validation failed",
+      );
+    }
+  };
+
+  const handleReject = () => {
+    request.resolve({ action: "reject" });
+    onClose();
+  };
```

Comment:
`reject` has been reverted to `decline`. It was slipped into the documentation at the last minute, but not the schema. When this was realized, the team decided to revert rather than make a post-release update to the schema.

```suggestion
  const handleDecline = () => {
    request.resolve({ action: "decline" });
    onClose();
  };
```